source/fundamentals/data-formats/document-data-format-extended-json.txt

```diff
@@ -164,47 +192,48 @@ The output of this code example resembles the following:
    507f1f77bcf86cd799439011 is type: org.bson.types.ObjectId
    4794261 is type: java.lang.Long
 
-To construct an instance of a ``Document`` from Extended JSON, you can call
-the ``parse()`` static factory method of the ``Document`` or ``BsonDocument``
-class. This method instantiates a ``JsonReader``, parses the Extended JSON
-string, and returns an instance of that class containing the data.
 
-The following example shows how you can use the ``Document`` class to read
-the same example Extended JSON string into a ``Document`` object using the
-``parse()`` method:
+For additional information on the classes and methods mentioned on this
+section, see the following API documentation:
 
-.. code-block:: java
+Write Extended JSON
+-------------------
 
-   String ejsonStr = "{ \"_id\": { \"$oid\": \"507f1f77bcf86cd799439011\"}," +
-                     "\"myNumber\": {\"$numberLong\": \"4794261\" }}}";
+Using the Document Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   Document doc = new Document(Document.parse(ejsonStr));
-   System.out.println(doc);
+You can write an Extended JSON string from an instance of ``Document`` or
+``BsonDocument`` by calling the ``toJson()`` method, passing it an instance of
+``JsonWriterSettings`` to specify the Extended JSON format.
 
-The output of the code above resembles the following:
+In this example, we output the Extended JSON in the "Shell" format.
 
-.. code-block:: none
-   :copyable: False
+.. code-block:: java
 
-   Document{{_id=507f1f77bcf86cd799439011, myNumber=4794261}}
+   Document myDoc = new Document();
+   myDoc.append("_id", new ObjectId("507f1f77bcf86cd799439012")).append("myNumber", 11223344);
 
-For more information on the driver document classes, see our Fundamentals page
-on :doc:`Documents </fundamentals/data-formats/documents>`.
+   JsonWriterSettings settings = JsonWriterSettings.builder().outputMode(JsonMode.SHELL).build();
+   System.out.println(doc.toJson(settings));
 
-For additional information on the classes and methods mentioned on this
-section, see the following API documentation:
+The output of this code example resembles the following:
 
-Write Extended JSON
--------------------
+.. code-block:: none
+   :copyable: false
+
+   {"_id": ObjectId("507f1f77bcf86cd799439011"), "myNumber": NumberLong(4794261)}
+
+Using the BSON Library
+~~~~~~~~~~~~~~~~~~~~~~
 
-You can convert Java objects into Extended JSON with the BSON library by using
-the :java-docs:`JsonWriter </apidocs/bson/org/bson/json/JsonWriter.html>`
-class. To construct an instance of ``JsonWriter``, pass a
-subclass of a Java ``Writer`` to specify how you want to output the Extended
-JSON. You can optionally pass a :java-docs:`JsonWriterSettings </apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+You can also read an Extended JSON string into Java objects using the BSON
```

Comment:
This is the opposite of what it does.  It writes JSON, not reads it.

Reply:
Addressed in https://github.com/mongodb/docs-java/pull/37/commits/47fea5dd750da5601c6ba188a3bf768aa5b9a954